pkg/cmd/provisioning/gcp/create_workload_identity_provider.go

```diff
@@ -0,0 +1,255 @@
+)
+
+func createWorkloadIdentityProviderCmd(cmd *cobra.Command, args []string) {
+	ctx := context.Background()
+
+	gcpClient, err := gcp.NewClient(CreateWorkloadIdentityProviderOpts.Project, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	publicKeyPath := CreateWorkloadIdentityProviderOpts.PublicKeyPath
+	if publicKeyPath == "" {
+		publicKeyPath = filepath.Join(CreateWorkloadIdentityProviderOpts.TargetDir, provisioning.PublicKeyFile)
+	}
+
+	err = createWorkloadIdentityProvider(ctx, gcpClient, CreateWorkloadIdentityProviderOpts.Name, CreateWorkloadIdentityProviderOpts.Region, CreateWorkloadIdentityProviderOpts.Project, CreateWorkloadIdentityProviderOpts.WorkloadIdentityPool, publicKeyPath, CreateWorkloadIdentityProviderOpts.TargetDir, CreateWorkloadIdentityProviderOpts.DryRun)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func createWorkloadIdentityProvider(ctx context.Context, client gcp.Client, name, region, project, workloadIdentityPool string, publicKeyPath, targetDir string, generateOnly bool) error {
+	// Create a storage bucket
+	bucketName := fmt.Sprintf("%s-oidc", name)
+	if err := createOIDCBucket(ctx, client, bucketName, region, project, targetDir, generateOnly); err != nil {
+		return err
+	}
+	issuerURL := fmt.Sprintf("https://storage.googleapis.com/%s", bucketName)
+
+	// Create the OIDC config file
+	if err := createOIDCConfiguration(ctx, client, bucketName, issuerURL, targetDir, generateOnly); err != nil {
+		return err
+	}
+
+	// Create the OIDC key list
+	if err := createJSONWebKeySet(ctx, client, publicKeyPath, bucketName, targetDir, generateOnly); err != nil {
+		return err
+	}
+
+	// Create the workload identity provider
+	err := createIdentityProvider(ctx, client, name, project, issuerURL, workloadIdentityPool, targetDir, generateOnly)
+	if err != nil {
+		return err
+	}
+
+	// Create the installer manifest file
+	if err := provisioning.CreateClusterAuthentication(issuerURL, targetDir); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createOIDCBucket(ctx context.Context, client gcp.Client, bucketName, region, project, targetDir string, generateOnly bool) error {
+	if generateOnly {
+		createOidcBucketScript := provisioning.CreateShellScript([]string{createOidcBucketCmd})
+		createOidcBucketScriptFilepath := filepath.Join(targetDir, provisioning.CreateOidcBucketScriptName)
+		script := fmt.Sprintf(createOidcBucketScript, region, project, bucketName)
+		err := provisioning.SaveToFile("shell script to create OIDC bucket", createOidcBucketScriptFilepath, []byte(script))
+		if err != nil {
+			return errors.Wrap(err, "failed to save shell script to create oidc bucket")
+		}
+	} else {
+		bucketAttrs := &storage.BucketAttrs{
+			Name:                     bucketName,
+			Location:                 region,
+			UniformBucketLevelAccess: storage.UniformBucketLevelAccess{Enabled: true},
```

Comment:
It appears to me that this bucket is non-public. And I don't see in the PutObject() calls bellow that the objects are being marked explicitly as public. How exactly does this work? Do the Google Storage Buckets not need to be public when setting up the Identity Provider?

Reply:
You are right! I missed it. I need to set public access on the bucket.

Reply:
the created bucket and objects will be publicly accessible now